.github/workflows/generate_changelog.yml

```diff
@@ -0,0 +1,50 @@
+name: Generate Changelog
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_tag:
+        description: 'Release tag of latest release'
+        required: true
+      trigger_secret:
+        description: 'Common Secret stored in both repos for Authentication'
+        required: true
+
+permissions:
+      contents: write
+
+jobs:
+  process-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Verify Trigger Source
+        env:
+          RECIEVED_SECRET: ${{ inputs.trigger_secret }}
+          EXPECTED_SECRET: ${{ secrets.CHANGELOG_TRIGGER_SECRET }}
+        run: |
+          if [ "$RECIEVED_SECRET" != "$EXPECTED_SECRET" ]; then
+            echo "Unauthorized trigger"
+            exit 1
+          fi
+
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      
+      - name: Install Dependencies
+        run: pip install -r requirements.txt
+          
+      - name: Run Script
+        env:
+          RELEASE_TAG: ${{ inputs.release_tag }}
+          GITHUB_PAT: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+        run: python scripts/create_changelog.py "$RELEASE_TAG"
+      
+      - name: Commit and Push Changes
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git add .
+          git commit -m "Added changelog for release: ${{ inputs.release_tag }}"
+          git push
```

Comment:
Could the user.name and user.email be simpler?

We may use a specific file name (`src/changelogs/sage-......txt`?) instead of `.` for safety?

Reply:
> Could the user.name and user.email be simpler?

Yes we could do that, but if we change the email then we won't see the black github-actions icon ![github-actions](https://avatars.githubusercontent.com/in/15368?s=20&v=4) in the commits. Instead we would see it in gray disabled mode.

>   We may use a specific file name (src/changelogs/sage-......txt?) instead of . for safety?

Yes, will fix that